sympy/functions/elementary/complexes.py

```diff
@@ -92,10 +107,35 @@ def eval(cls, arg):
     def as_real_imag(self, deep=True, **hints):
         """
         Returns the real number with a zero imaginary part.
+
+        Examples
+        ========
+
+        >>> from sympy import re, I
+        >>> from sympy.abc import x, y
+        >>> re(10 + 20*I).as_real_imag()
+        (10, 0)
+        >>> re(10).as_real_imag()
+        (10, 0)
+        >>> re(40*I).as_real_imag()
+        (0, 0)
+        >>> re(x+y*I).as_real_imag()
+        (re(x) - im(y), 0)
```

Comment:
I'd also find that this example is not intuitive because people may expect this ti give `x` rather than `re(x) - im(y)` in elementary complex algebra setting.
So I'd better state about assumptions `x, y = symbols('x y', real=True)`

Reply:
Done.